e2etest/bgptests/unnummbered.go

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"time"
+
+	"go.universe.tf/e2etest/pkg/config"
+	"go.universe.tf/e2etest/pkg/container"
+	"go.universe.tf/e2etest/pkg/executor"
+	"go.universe.tf/e2etest/pkg/frr"
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	frrcontainer "go.universe.tf/e2etest/pkg/frr/container"
+	jigservice "go.universe.tf/e2etest/pkg/jigservice"
+	"go.universe.tf/e2etest/pkg/netdev"
+	testservice "go.universe.tf/e2etest/pkg/service"
+
+	"go.universe.tf/e2etest/pkg/k8s"
+	"go.universe.tf/e2etest/pkg/k8sclient"
+	"go.universe.tf/e2etest/pkg/metallb"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
+	metallbv1beta2 "go.universe.tf/metallb/api/v1beta2"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+var FRRImage string
+var _ = ginkgo.Describe("FRRK8S-MODE Unnumbered BGP", func() {
+	var (
+		testNamespace         string
+		nodeWithP2PConnection corev1.Node
+		externalP2PContainer  *frrcontainer.FRR
+
+		cs clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		ginkgo.By("Clearing any previous configuration")
+		err := ConfigUpdater.Clean()
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = k8sclient.New()
+		allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		nodeWithP2PConnection = allNodes.Items[0]
+
+		err = k8s.DeleteNamespace(k8sclient.New(), "unnumbered-bgp")
+		Expect(err).NotTo(HaveOccurred())
+		testNamespace, err = k8s.CreateTestNamespace(k8sclient.New(), "unnumbered-bgp")
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := k8s.DeleteNamespace(k8sclient.New(), testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentSpecReport().Failed() {
+			k8s.DumpInfo(Reporter, ginkgo.CurrentSpecReport().LeafNodeText)
+			dumpBGPInfo(ReportPath, ginkgo.CurrentSpecReport().LeafNodeText, cs, testNamespace, []*frrcontainer.FRR{externalP2PContainer})
+		}
+	})
+
+	ginkgo.DescribeTable("Session is established and route is advertised", func(prefixSend []string, iface string, tweak func(svc *corev1.Service)) {
+		rc := frrconfig.RouterConfigUnnumbered{
+			ASNLocal:  metalLBASN,
+			ASNRemote: metalLBASN,
+			Hostname:  "tor1",
+			Interface: iface,
+			RouterID:  "1.1.1.1",
+		}
+
+		var err error
+		ginkgo.By(fmt.Sprintf("creating p2p %s:%s -- %s:external-p2p-container", nodeWithP2PConnection.Name, iface, iface))
+		externalP2PContainer, err = frrcontainer.CreateP2PPeerFor(nodeWithP2PConnection.Name, iface, FRRImage)
```

Comment:
externalP2P should be local and not local. Also, removing the container in defer is fragile, compared to having it done in an `AfterSomething`

Reply:
Done 

Can you please expand why the defer is fragile?

Reply:
because if you ctrl-c the test execution, you'll lose the defer while the after is always executed (unless you ctrl-c / ctrl-c)